### PR TITLE
Add table `gcp_compute_instance_group` Closes #400

### LINKED ISSUE #400: Add table gcp_compute_instance_group
**References**

TF Ref - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance_group
API Ref - https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups/list

### PR DESCRIPTION
**Pull request by karanpopat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Outputs:

project = "parker-aaa"
resource_aka = "gcp://compute.googleapis.com/projects/parker-aaa/zones/us-central1-a/instanceGroups/turbottest14683"
resource_id = "projects/parker-aaa/zones/us-central1-a/instanceGroups/turbottest14683"
resource_name = "turbottest14683"
self_link = "https://www.googleapis.com/compute/v1/projects/parker-aaa/zones/us-central1-a/instanceGroups/turbottest14683"
size = 0

Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "gcp://compute.googleapis.com/projects/parker-aaa/zones/us-central1-a/instanceGroups/turbottest14683"
    ],
    "description": "turbottest14683",
    "location": "us-central1-a",
    "name": "turbottest14683",
    "project": "parker-aaa",
    "self_link": "https://www.googleapis.com/compute/v1/projects/parker-aaa/zones/us-central1-a/instanceGroups/turbottest14683",
    "size": "0",
    "title": "turbottest14683"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "gcp://compute.googleapis.com/projects/parker-aaa/zones/us-central1-a/instanceGroups/turbottest14683"
    ],
    "description": "turbottest14683",
    "location": "us-central1-a",
    "name": "turbottest14683",
    "project": "parker-aaa",
    "self_link": "https://www.googleapis.com/compute/v1/projects/parker-aaa/zones/us-central1-a/instanceGroups/turbottest14683",
    "size": "0",
    "title": "turbottest14683"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "akas": [
      "gcp://compute.googleapis.com/projects/parker-aaa/zones/us-central1-a/instanceGroups/turbottest14683"
    ],
    "description": "turbottest14683",
    "location": "us-central1-a",
    "name": "turbottest14683",
    "project": "parker-aaa",
    "self_link": "https://www.googleapis.com/compute/v1/projects/parker-aaa/zones/us-central1-a/instanceGroups/turbottest14683",
    "size": "0",
    "title": "turbottest14683"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

POSTTEST: tests/gcp_compute_instance_group

TEARDOWN: tests/gcp_compute_instance_group

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```sql
> select 
  name,
  description,
  size,
  location, 
  akas, 
  project
from 
  gcp_compute_instance_group;
+------------------+------------------+------+---------------+----------------------------------------------------------------------------------------------------------+------------+
| name             | description      | size | location      | akas                                                                                                     | project    |
+------------------+------------------+------+---------------+----------------------------------------------------------------------------------------------------------+------------+
| instance-group-1 | instance-group-1 | 1    | us-central1-a | ["gcp://compute.googleapis.com/projects/parker-aaa/zones/us-central1-a/instanceGroups/instance-group-1"] | parker-aaa |
+------------------+------------------+------+---------------+----------------------------------------------------------------------------------------------------------+------------+

> select
  g.name,
  ins.name as instance_name,
  ins.status as instance_status
from
  gcp_compute_instance_group as g,
  jsonb_array_elements(instances) as i,
  gcp_compute_instance as ins
where
  (i ->> 'instance') = ins.self_link;
+------------------+---------------+-----------------+
| name             | instance_name | instance_status |
+------------------+---------------+-----------------+
| instance-group-1 | instance-1    | RUNNING         |
+------------------+---------------+-----------------+

```
</details>

### CODE REVIEW

File: docs/tables/gcp_compute_instance_group.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  size,
+  location, 
+  akas, 
+  project
+from 
+  gcp_compute_instance_group;
+```
+
+### Get number of instances per instance group
+```sql
```
Comment (misraved) [inline docs/tables/gcp_compute_instance_group.md:23]:
```suggestion
### Get number of instances per instance group

```sql
```

File: docs/tables/gcp_compute_instance_group.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+select
+  name,
+  size as no_of_instances
+from
+  gcp_compute_instance_group;
+```
+
+### Get instance details of each instance group
+```sql
```
Comment (misraved) [inline docs/tables/gcp_compute_instance_group.md:32]:
```suggestion
### Get instance details of each instance group

```sql
```

File: gcp-test/tint.js [hunk 3]
```diff
@@ -384,7 +384,7 @@ const _runGraphqlQuery = function (test, query) {
   return new Promise((resolve, reject) => {
     try {
       var queryTmp = _renderToTmp(test, query.query);
-      var variablesTmp = _renderToTmp(test, query.variables, {});
+      // var variablesTmp = _renderToTmp(test, query.variables, {});
```
Comment (misraved) [inline gcp-test/tint.js:387]: Please revert back the change 👍 .

File: gcp/table_gcp_compute_instance_group.go [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("name"),
+			Hydrate:    getComputeInstanceGroup,
+		},
+		List: &plugin.ListConfig{
+			Hydrate:           listComputeInstanceGroup,
+			ShouldIgnoreError: isIgnorableError([]string{"403"}),
+		},
+		Columns: []*plugin.Column{
+			// commonly used columns
```
Comment (misraved) [inline gcp/table_gcp_compute_instance_group.go:27]: Is this comment needed?

File: gcp/table_gcp_compute_instance_group.go [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+	projectId, err := getProjectCached(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	project := projectId.(string)
+
+	// Create Service Connection
+	service, err := ComputeService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Debug("gcp_compute_instance_group.getComputeInstanceGroupInstances", "service_creation_err", err)
```
Comment (misraved) [inline gcp/table_gcp_compute_instance_group.go:260]:
```suggestion
		plugin.Logger(ctx).Error("gcp_compute_instance_group.getComputeInstanceGroupInstances", "service_creation_err", err)
```
Should this be error or debug?

File: gcp/table_gcp_compute_instance_group.go [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+	zoneName := getLastPathElement(types.SafeString(i.Zone))
+	regionName := getLastPathElement(types.SafeString(i.Region))
+	project := strings.Split(i.SelfLink, "/")[6]
+	instanceGroupName := types.SafeString(i.Name)
+
+	akas := []string{"gcp://compute.googleapis.com/projects/" + project + "/zones/" + zoneName + "/instanceGroups/" + instanceGroupName}
+
+	if zoneName == "" {
+		akas = []string{"gcp://compute.googleapis.com/projects/" + project + "/regions/" + regionName + "/instanceGroups/" + instanceGroupName}
+	}
```
Comment (misraved) [inline gcp/table_gcp_compute_instance_group.go:288]: Can we compose this into a single `if-else` block?

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments, thanks!!

File: gcp-test/tests/gcp_compute_instance_group/test-get-expected.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "akas": [
+      "{{ output.resource_aka.value }}"
+    ],
+    "description": "{{resourceName}}",
```
Comment (ParthaI) [inline gcp-test/tests/gcp_compute_instance_group/test-get-expected.json:6]:
```suggestion
    "description": "{{ resourceName }}",
```

File: gcp-test/tests/gcp_compute_instance_group/test-get-expected.json [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    "location": "us-central1-a",
+    "name": "{{resourceName}}",
```
Comment (ParthaI) [inline gcp-test/tests/gcp_compute_instance_group/test-get-expected.json:8]:
```suggestion
    "name": "{{ resourceName }}",
```

File: gcp/table_gcp_compute_instance_group.go [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+
+			// zone_name is a simpler view of the zone, without the full path
+			{
+				Name:        "zone_name",
+				Description: "The zone name in which the instance group resides.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Zone").Transform(lastPathElement),
+			},
+
+			// standard steampipe columns
```
Comment (ParthaI) [inline gcp/table_gcp_compute_instance_group.go:117]:
```suggestion
			// Steampipe standard columns
```

File: gcp/table_gcp_compute_instance_group.go [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(instanceGroupAka),
+			},
+
+			// standard gcp columns
```
Comment (ParthaI) [inline gcp/table_gcp_compute_instance_group.go:131]:
```suggestion
			// GCP standard columns
```

**Review by ParthaI — COMMENTED** [review]
@karanpopat, please see the comments, and make changes where ever required. Thanks!

**Review by ParthaI — APPROVED** [review]
LGTM

**Pull request merged**